lmms_eval/tasks/mme_realworld/utils.py

```diff
@@ -0,0 +1,206 @@
+import datetime
+import json
+import os
+import re
+import sys
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import cv2
+import numpy as np
+import yaml
+from loguru import logger as eval_logger
+
+from lmms_eval.tasks._task_utils.file_utils import generate_submission_file
+
+TASKS = [
+    'Reasoning',
+    'Perception',
+]
+
+SUBTASKS = [
+    'Monitoring',
+    'Autonomous_Driving',
+    'OCR with Complex Context',
+    'Diagram and Table',
+    'Remote Sensing',
+]
+
+replace_prompt = " Please answer yes or no."
+
+# with open(Path(__file__).parent / "_default_template_yaml", "r") as f:
+#     raw_data = f.readlines()
+#     safe_data = []
+#     for i, line in enumerate(raw_data):
+#         # remove function definition since yaml load cannot handle it
+#         if "!function" not in line:
+#             safe_data.append(line)
+
+#     config = yaml.safe_load("".join(safe_data))
+
+hf_home = os.getenv("HF_HOME", "~/.cache/huggingface/")
+# cache_dir = os.path.join(hf_home, cache_dir)
+# base_cache_dir = config["dataset_kwargs"]["cache_dir"]
+base_cache_dir = os.path.expanduser(hf_home)
+with open(Path(__file__).parent / "mme_realworld.yaml", "r") as f:
```

Comment:
If your dataset is purely image and does not rely on gpt-eval, there is no need to load the yaml config from here